sqlmesh/core/table_diff.py

```diff
@@ -223,15 +281,49 @@ def row_diff(self) -> RowDiff:
                     )
                     .limit(self.limit)
                 )
+                sample = self.adapter.fetchdf(sample_query)
+
+                joined_sample_cols = [f"s__{c}" for c in index_cols]
+                comparison_cols = [
+                    (f"s__{c}", f"t__{c}")
+                    for c in column_stats[column_stats["pct_match"] < 100].index
+                ]
+                for cols in comparison_cols:
+                    joined_sample_cols.extend([*cols])
+                joined_sample = sample[sample["rows_joined"] == 1][joined_sample_cols]
+                joined_sample.rename(
+                    columns={
+                        c: c.split("__")[1] if c.split("__")[1] in index_cols else c
+                        for c in joined_sample_cols
+                    },
+                    inplace=True,
+                )
+
+                s_sample = sample[(sample["s_exists"] == 1) & (sample["rows_joined"] == 0)][
+                    [
+                        *[f"s__{c}" for c in index_cols],
+                        *[f"s__{c}" for c in column_stats.index if not c in index_cols],
+                    ]
+                ]
+
+                t_sample = sample[(sample["t_exists"] == 1) & (sample["rows_joined"] == 0)][
+                    [
```

Comment:
is the nested array correct?

Reply:
good catch

Reply:
assuming the source of non-index cols was what you were commenting on